integration/helpers.go

```diff
@@ -20,8 +20,90 @@ import (
 
 	"github.com/stretchr/testify/require"
 	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/bson/primitive"
+
+	"github.com/FerretDB/FerretDB/internal/types"
+	"github.com/FerretDB/FerretDB/internal/util/testutil"
 )
 
+// convert converts given driver value (bson.D, bson.A, etc) to FerretDB types package value.
+//
+// It then can be used with all types helpers such as testutil.AssertEqual.
+func convert(t testing.TB, v any) any {
+	t.Helper()
+
+	switch v := v.(type) {
+	// composite types
+	case primitive.D:
+		doc := types.MustNewDocument()
+		for _, e := range v {
+			doc.Set(e.Key, convert(t, e.Value))
+		}
+		return doc
+	case primitive.A:
+		arr := types.MakeArray(len(v))
+		for _, e := range v {
+			arr.Append(convert(t, e))
+		}
+		return arr
+
+	// scalar types (in the same order as in types package)
+	case float64:
+		return v
+	case string:
+		return v
+	case primitive.Binary:
+		return types.Binary{
+			Subtype: types.BinarySubtype(v.Subtype),
+			B:       v.Data,
+		}
+	case bool:
+		return v
+	case primitive.DateTime:
+		return v.Time()
+	case nil:
+		return v
+	case primitive.Regex:
+		return types.Regex{
+			Pattern: v.Pattern,
+			Options: v.Options,
+		}
+	case int32:
+		return v
+	case primitive.Timestamp:
+		return types.Timestamp(uint64(v.I)<<32 + uint64(v.T))
+	case int64:
+		return v
+	default:
+		t.Fatalf("unexpected type %T", v)
+		panic("not reached")
```

Comment:
test fatal should fail, isn't it? it's not fails?

Reply:
It does fail, but compile is not smart enough to understand that return is not needed there
```
./helpers.go:81: missing return
```

I wanted to be sure that all cases have returns, so panic acts there as impossible [terminal statement](https://go.dev/ref/spec#Terminating_statements)